src/Index.jsx

```diff
@@ -1,25 +1,59 @@
+/* global TERRA_DEV_SITE_BASENAME */
+// TERRA_DEV_SITE_BASENAME is defined by webpack
 import React from 'react';
 import ReactDOM from 'react-dom';
-import { HashRouter as Router } from 'react-router-dom';
+
+import ApplicationBase from 'terra-application';
+import ApplicationNavigation from 'terra-application-navigation';
+
 // This line will be resolved by webpack
 // eslint-disable-next-line import/no-unresolved, import/extensions
-import siteConfig from 'build/appConfig';
+import siteConfig from 'build/siteConfig';
 
-import App from './app/App';
+import Site from './site/_Site';
+
+// eslint-disable-next-line import/no-unresolved, import/extensions
+const fetchSearchItems = () => import(/* webpackPrefetch: true, webpackChunkName: "build/searchItems" */ 'build/searchItems').then(({ default: items }) => items);
 
-const Site = () => (
-  <Router>
-    <App
-      nameConfig={siteConfig.nameConfig}
-      utilityConfig={siteConfig.utilityConfig}
-      routingConfig={siteConfig.routingConfig}
-      navigationItems={siteConfig.navigationItems}
-      extensions={siteConfig.extensions}
-      indexPath={siteConfig.indexPath}
-      defaultTheme={siteConfig.defaultTheme}
-      themes={siteConfig.themes}
-    />
-  </Router>
+const TerraDevSite = () => (
+  <Site
+    applicationBase={({ locale, themeName, child }) => (
```

Comment:
The `applicationBase` and `applicationNavigation` props seem really odd to me. Especially since they are functions with expected props to be provided that aren't defined in this file?

Could we move this responsibility to `_Site.jsx`? to render ApplicationBase / ApplicationNavigation directly? 
or 
Create new dev-site components `_ApplicationBase.jsx` & `_ApplicationNavigation`?

Reply:
We're using render props in index.jsx for future x_dev_site's to be able to reuse the overall framework and swap out just the pieces they need TerraApplicationBase and TerraApplicationNavigation.